docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,132 @@
+# Integration tests
+
+There are, currently, three types of Gosoline integration tests, each of them meant to be used for a specific situation. For all of them, the Gosoline `suite` package is responsible for reading their configuration, starting and running an application, then running the actual tests cases. Below is their description:
+
+### resty.Client integration tests
+
+In the [Integration tests for your API](../getting_started/integration_tests.md) example, we saw the following test:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_ToEuro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_ToEuro(_ suite.AppUnderTest, client *resty.Client) error {
+	var result float64
+
+	response, err := client.R().
+		SetResult(&result).
+		Execute(http.MethodGet, "/euro/10/GBP")
+
+	s.NoError(err)
+	s.Equal(http.StatusOK, response.StatusCode())
+	s.Equal(8.0, result)
+
+	return nil
+}
+```
+
+The `Test_ToEuro` method makes a GET call to an endpoint, in order to receive back the euro exchange value for 10 GBP. Lastly, it checks if the received value is 8.0.
+
+In order to make this GET call `Test_ToEuro` does not need to concern itself about the IP on which the exchange application is running, nor its port. All `Test_ToEuro` needs to know is the URL path of that endpoint, as the _client_ object does the rest.
+
+This _client_ object is provided out-of-the box by Gosoline, whenever a test is defined like this:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupSuite/ /return definer.ApiDefiner\n}/)
+```go
+func (s *ApiTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("info"),
+		suite.WithConfigFile("../api/config.dist.yml"),
+		suite.WithFixtures(fixtureSets),
+		suite.WithClockProvider(s.clock),
+	}
+}
+
+func (s *ApiTestSuite) SetupApiDefinitions() apiserver.Definer {
+	return definer.ApiDefiner
+}
+```
+
+Implementing _SetupApiDefinitions_ when configuring your test will inform Gosoline that an API is being tested, thus the _client_ object will contain meaningful data. The `SetupSuite` method is shown to illustrate how similar is the configuration for all three types of tests.
+
+Forgeting to implement _SetupApiDefinitions_, and the trying to run an _resty.Client_ test case, will result in an error reminding you that `the suite has to implement the TestingSuiteApiDefinitionsAware interface`.
+
+### ApiServer integration tests
+
+The _client_ object is permissive, allowing access to the request itself. A simpler way to test an API is provided by Gosoline's _ApiServerTestCase_. These type of tests also issue HTTP calls and compare their responses against a predefined value, but they need to follow a predefined structure. Below we can see how the _resty client_ test above would look like as an _ApiServerTestCase_:
```

Comment:
I think we should highlight the main advantage of the client object compared to using the `ApiServerTestCase` - it allows you to control when requests are executed and gives you access to the endpoint (by providing the host+port), so if you really want to time requests or similar, this is the way to go. If you just make a standard API call and read the response, there is no need for that and you can use the declarative `ApiServerTestCase`